### ISSUE TITLE
Formatting on save does not work with heex files.

### ISSUE THREAD

**Issue by tlietz** [elixir-lsp/vscode-elixir-ls#242]
The auto formatter does not work on `heex` files, even after setting up the configuration in my project correctly and adding the elixir-ls plugin as the default formatter in `settings.json`. If I run a `mix format` in the terminal, the `heex` files are formatted.

```elixir
# .formatter.exs
[
  import_deps: [:ecto, :phoenix],
  plugins: [Phoenix.LiveView.HTMLFormatter],
  inputs: ["*.{heex,ex,exs}", "priv/*/seeds.exs", "{config,lib,test}/**/*.{heex,ex,exs}"],
  subdirectories: ["priv/*/migrations"]
]
```

```elixir
# settings.json
 "[phoenix-heex]": {
        "editor.formatOnSave": true,
        "editor.defaultFormatter": "JakeBecker.elixir-ls"
    },
```

Is this expected? Is it something to implement on the vscode-elixir-ls level?

### Environment

- Elixir & Erlang versions (elixir --version):  Erlang/OTP 24, Elixir 1.13.4
- VSCode ElixirLS version: 0.9.0
- Operating System Version: macOS Monterey 12.1 (M1)

**Comment by dvic** [comment]
I have the same issue, error is

```
[Error - 7:51:17 AM] Request textDocument/formatting failed.
  Message: Unable to format due to syntax error
  Code: -32603
```

**Comment by dvic** [comment]
Just confirmed: latest master version of this repo + latest master of elixir-ls suffers from same issue.

**Comment by holandes22** [comment]
Same for me

```[Error - 9:17:19 AM] Request textDocument/formatting failed.
  Message: Unable to format due to syntax error
  Code: -32603 
```

I installed latest elixir 1.13.4 and liveview. Automatic formatting in vscode works great for `~H` sigil, but fails on `.heex` files.
I tried formatting from the command line directly the `.heex` files and that work ok for me.

Not sure if related, but I noticed in the elixir ls output that it says is using version 0.8.2

```
Started ElixirLS v0.8.2
Elixir version: "1.13.4 (compiled with Erlang/OTP 22)"
Erlang version: "24"
ElixirLS compiled with Elixir 1.10.4 and erlang 22
```

But there is not release with that number in the [changelog](https://github.com/elixir-lsp/elixir-ls/blob/master/CHANGELOG.md#v090-4-december-2021) as it jumps from 0.8.1 to 0.9.0 so maybe that is just the latest release

**Comment by dvic** [comment]
> Same for me
> 
> ```
>   Message: Unable to format due to syntax error
>   Code: -32603 
> ```
> 
> I installed latest elixir 1.13.4 and liveview. Automatic formatting in vscode works great for `~H` sigil, but fails on `.heex` files. I tried formatting from the command line directly the `.heex` files and that work ok for me.
> 
> Not sure if related, but I noticed in the elixir ls output that it says is using version 0.8.2
> 
> ```
> Started ElixirLS v0.8.2
> Elixir version: "1.13.4 (compiled with Erlang/OTP 22)"
> Erlang version: "24"
> ElixirLS compiled with Elixir 1.10.4 and erlang 22
> ```
> 
> But there is not release with that number in the [changelog](https://github.com/elixir-lsp/elixir-ls/blob/master/CHANGELOG.md#v090-4-december-2021) as it jumps from 0.8.1 to 0.9.0 so maybe that is just the latest release

I see the same version mismatch, but I'm not sure if it matters, even with the custom build (everything bumped to master), where I do see the correct version 

```
Started ElixirLS v0.9.0
Elixir version: "1.13.3 (compiled with Erlang/OTP 24)"
Erlang version: "24"
ElixirLS compiled with Elixir 1.13.4 and erlang 24
```

it doesn't work.

**Comment by holandes22** [comment]
Yeah, I tried that as well (compiling from master). Error remains

**Comment by holandes22** [comment]
I tried reproducing the error while running the formatter in the command line. I installed with asdf a different elixir version (1.12.0) and then I get a similar syntax error when formatting a `.heex` file

```
mix format failed for file: edit.html.heex
** (SyntaxError) edit.html.heex:1:1: syntax error before: '<'
    (elixir 1.12.0) lib/code.ex:978: Code.format_string!/2
    (mix 1.12.0) lib/mix/tasks/format.ex:418: Mix.Tasks.Format.format_file/2
    (elixir 1.12.0) lib/task/supervised.ex:90: Task.Supervised.invoke_mfa/2
    (elixir 1.12.0) lib/task/supervised.ex:35: Task.Supervised.reply/5
    (stdlib 3.17.1) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
```

so maybe is related to the plugin using an older Elixir version, but what I don't get, if that is the case, how it does work on `~H` sigils

**Comment by Nicd** [comment]
As a workaround, you can install the runonsave extension for VSCode and configure that to run `mix format`: https://elixirforum.com/t/heex-formatter-a-formatter-for-heex-templates/45649/18?u=nicd

Sadly then "save without formatting" won't work. So it would be nice to get this working with the extension directly.

**Comment by cchauche** [comment]
I wanted to offer an alternative solution that worked for me. I installed another VSCode extension called [Elixir Mix Formatter](https://marketplace.visualstudio.com/items?itemName=animus-coop.vscode-elixir-mix-formatter) and set this to be my default formatter for elixir and heex files which seems to work as expected.

**Comment by dvic** [comment]
https://github.com/elixir-lsp/elixir-ls/pull/690 just got merged... 🥳

**Comment by benvp** [comment]
Just in case anybody wonders how to compile the current master, try this little snippet:

```bash
git clone --depth 1 --branch master https://github.com/elixir-lsp/elixir-ls.git ~/elixir_ls
cd ~/elixir_ls
mix deps.get
mix elixir_ls.release -o ~/.vscode/extensions/jakebecker.elixir-ls-0.10.0/elixir-ls-release/
```

**Comment by trkoch** [comment]
For everyone tracking this, note `elixir-ls` was updated in with version 0.11.0 of the extension, according to [elixir-ls/CHANGELOG.md](https://github.com/elixir-lsp/elixir-ls/blob/fe1191031874bf0d05afa18d1886e8ac705fd94d/CHANGELOG.md#v0110-14-august-2022:~:text=Added%20support%20for%20mix%20formatter%20plugins%20(Dalibor%20Horinek)).

**Comment by lukaszsamson** [comment]
Fixed in v0.11
